util/microservice-sdk/js/README.md

```diff
@@ -8,6 +8,131 @@ library scoped to attend the dojot microservices necessities.
+* message: message in string format
+* extra: any additional data aggregated in an object - user specific (optional metadada)
+
+The messages are logged in the 'file' using the following json format:
+
+```js
+{
+  level: string,
+  timestamp: <number>,
+  sid: <string>,
+  rid: <string>,
+  message: <string>,
+  file: <string>,
+  line: <number>,
+  extra: <object>,
+ }
+```
+
+Where:
+
+* ts: timestamp in unix format
+* sid: service identification
+* level: error, warn, info or debug
+* rid: request identifier (optional metadata)
+* message: message in string format
+* file: file of the log message (optional metadata)
+* line: line of the log message (optional metadata)
+* extra: any additional data aggregated in an object - user specific (optional metadada)
+
+The transports ('console' and 'file') are shared by all instances of the wrapper. So,
+any change in the transports will affect all instances.
+
+By default, no transport is set, so it is necessary to do this as part of the configuration
+or initialization of the microservice.
+
+It's usage is very simple and is shown by the following example:
+
+```js
+const { Logger } = require('@dojot/microservice-sdk');
+
+// By default, no transport is enabled; so you need to
+// set at least one before to log the messages.
+// It's worth to say that the transports are shared by all
+// modules of the microservice; consequently, any change
+// in the configuration will be valid for all!
+
+// Setting console transport
+// For more information about it, see:
+// https://github.com/winstonjs/winston/blob/HEAD/docs/transports.md#console-transport
+//
+// This is the expected transport to be used in a Docker container
+// once it is stateless.
+Logger.setTransport('console', {
+  // Any configuration put here will be merged with the defaults:
+  // { level: 'info',
+  //   format: consoleFormat /* customized format */}
+  level: 'debug',
+});
+
+// Setting file transport
+// For more information about it, see:
+// https://github.com/winstonjs/winston-daily-rotate-file#readme
+//
+// This transport should be used if you need to keep the logs in
+// files with rotation.
+// This is not typically the case for Docker container applications,
+// where the logs are expected to be redirected to /dev/stdout
+Logger.setTransport('file', {
+  // Any configuration put here will be merged with the defaults:
+  // { level: 'info',
+  //   dirname: '/var/log/',
+  //   filename: 'dojot.microservice-%DATE%.log',
+  //   datePattern: 'YYYY-MM-DD',
+  //   zippedArchive: true,
+  //   maxSize: '10m',
+  //   maxFiles: '7d',
+  //   format: fileFormat, /* customized format */
+  //   level: 'debug',
+  // }
+  level: 'debug',
+  filename: 'sample-app-%DATE%.log',
+});
+
+// Enables the verbose mode. This should be avoided in
+// production and only used for debugging purposes.
+// When enabled, file:line information is added to each
+// logging message
+Logger.setVerbose(true);
```

Comment:
This approach was much better than the previous one, but I believe that it is still not ideal because to activate/deactivate this behavior it is necessary to change the code and this is something practically impossible in production (for debugging purposes).
We are also at risk of the developer forgetting to remove the snippet that activates this behavior (`Logger.setVerbose (true)`) and the code to be released for production. As we must avoid that this behavior is the default in production, I believe that a better approach is to use environment variables.
[Example of how Express.js works.](http://expressjs.com/en/guide/debugging.html)

Reply:
I don't see this a problem.  The setVerbose() is just a library method and can be called at any moment to change the logging mode. Furthermore, the service built with the SDK can provide mechanisms to enable or disable verbose mode through configuration or environment variables like Express.js